R/format_value.R

```diff
@@ -61,18 +62,22 @@ format_value.logical <- format_value.numeric
 
 #' @importFrom stats na.omit
 #' @keywords internal
-.format_value_unless_integer <- function(x, digits = 2, ...) {
+.format_value_unless_integer <- function(x, digits = 2, na = "", ...) {
   if (is.numeric(x) && !all(is.int(stats::na.omit(x)))) {
-    .format_value(x, digits = digits)
+    .format_value(x, digits = digits, na = na)
+  } else if (is.na(x)) {
+    as.character(na)
```

Comment:
Just a comment, we need `as.character()` here, because `NA` can be numeric, or character, or... So we need to make sure to have a "character"-`NA` if the user sets `missing = NA`.